src/scripts/Core.js

```diff
@@ -76,21 +76,15 @@ class Core {
       key.val = new_val;
     }
     else {
-      if (property.keys.length === 0) {
```

Comment:
I don't think it makes sense that just update `val` of property when `property.keys.length === 0`, so changed it so that simply add the new value to keys. Please let me know if it is not right.